zone/command.cpp

```diff
@@ -14106,6 +14107,134 @@ void command_network(Client *c, const Seperator *sep)
 	}
 }
 
+void command_viewzoneloot(Client *c, const Seperator *sep)
+{
+	std::map<uint32,ItemList> zone_loot_list;
+	auto npc_list = entity_list.GetNPCList();
+	uint32 loot_amount = 0, loot_id = 1, search_item_id = 0;
+	if (sep->argnum == 1 && sep->IsNumber(1)) {
+		search_item_id = atoi(sep->arg[1]);
+	} else if (sep->argnum == 1 && !sep->IsNumber(1)) {
+		c->Message(
+			Chat::Yellow,
+			"Usage: #viewzoneloot [item id]"
+		);
+		return;
+	}
+	for (auto npc_entity : npc_list) {
+		auto current_npc_item_list = npc_entity.second->GetItemList();
+		zone_loot_list.insert({ npc_entity.second->GetID(), current_npc_item_list });
+	}
+	for (auto loot_item : zone_loot_list) {
+		uint32 current_entity_id = loot_item.first;
+		auto current_item_list = loot_item.second;
+		auto current_npc = (
```

Comment:
This can just be `auto current_npc = entity_list.GetNPCByID(current_entity_id);` since it returns `nullptr` on its own when it fails to find the NPC

You're making two calls to an entity list iteration because you have to check for it once and then fetch it once its valid versus just letting `GetNPCByID` return `nullptr`